msetnxex.lua

```diff
@@ -0,0 +1,11 @@
+local setExCount = 0
+if redis.call("msetnx", unpack(KEYS)) == 1 then
+    for i, key in ipairs(KEYS) do
+        if i % 2 == 1 then
```

Comment:
Right, this is not right. You should not mix values with keys. Your `KEYS` should only contains they keys. The `value` should be `ARGV[1]` and your `ttl` then `ARGV[2]`

Reply:
Ah ! yes it is kinda strange, but it keep the script pretty simple and it feels more optimized that way (not having to reconstruct a new array with KEYS + ARGS[1] content).
One comment to explain it better would to the trick perhaps ? Else I'l try to find a way to merge two arrays together...

Reply:
sorry, I appreciate the neatness of the code, but keys must be passed as keys to maintain long-term compatibility with redis and be compatible with cluster. if you pass a key as an argument, a redis cluster will not know where to find it.

Reply:
You are absolutly right, I updated the code accordingly.